lib/python/rose/suite_engine_proc.py

```diff
@@ -197,28 +197,40 @@ class SuiteStillRunningError(Exception):
 
     """An exception raised when a suite is still running."""
 
-    FMT_HEAD = "Suite \"%(suite_name)s\" may still be running.\n"
-    FMT_TAIL = "Try \"rose suite-shutdown --name=%(suite_name)s\" first?"
-    FMT_BODY1 = "Host \"%(host)s\" has %(reason_key)s:\n"
-    FMT_BODY2 = "    %(reason_value)s\n"
+    FMT_HEAD = "Suite \"%(suite_name)s\" has running processes on:"
+    FMT_BODY_1 = " %(host_name)s"
+    FMT_BODY_M = "\n    %(host_name)s"
+    FMT_TAIL = "\nTry \"rose suite-shutdown --name=%(suite_name)s\" first?"
 
     def __str__(self):
-        suite_name, reasons = self.args
+        suite_name, host_names = self.args
         ret = self.FMT_HEAD % {"suite_name": suite_name}
-        host = None
-        reason_key = None
-        for reason in reasons:
-            reason_dict = dict(reason)
-            if (reason_dict["host"] != host or
-                    reason_dict["reason_key"] != reason_key):
-                ret += self.FMT_BODY1 % reason_dict
-                host = reason_dict["host"]
-                reason_key = reason_dict["reason_key"]
-            ret += self.FMT_BODY2 % reason_dict
+        if len(host_names) == 1:
+            ret += self.FMT_BODY_1 % {"host_name": host_names[0]}
+        else:
+            for host_name in host_names:
+                ret += self.FMT_BODY_M % {"host_name": host_name}
         ret += self.FMT_TAIL % {"suite_name": suite_name}
         return ret
 
 
+class SuiteHostConnectError(Exception):
+    """Cannot connect to the suite host."""
+
+    TMPL = "%s: cannot connect to some suite hosts, try again later?\n"
```

Comment:
Wouldn't put a question mark on the end of this as its not used in any prompts, rather it is used when raising an exception.